Minor HTMLFrameOwnerElement cleanup

annevk opened the pull request:
#### 69fa6f6168e13a6a2b3b24f4b77ae2b0490b7dca
<pre>
Minor HTMLFrameOwnerElement cleanup
<a href="https://bugs.webkit.org/show_bug.cgi?id=268453">https://bugs.webkit.org/show_bug.cgi?id=268453</a>

Reviewed by Chris Dumez.

This makes the following changes:

- Move to dynamicDowncast&lt;HTMLFrameOwnerElement&gt; where applicable.
- Move to is&lt;HTMLFrameOwnerElement&gt; where applicable.
- Rename isFrameOwnerElement() to isHTMLFrameOwnerElement() for
  clarity.

* Source/WebCore/dom/Node.cpp:
(WebCore::showSubTreeAcrossFrame):
* Source/WebCore/dom/Node.h:
(WebCore::Node::isHTMLFrameOwnerElement const):
(WebCore::Node::isFrameOwnerElement const): Deleted.
* Source/WebCore/html/HTMLFrameOwnerElement.h:
(isType):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::unbind):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::findFocusCandidateInContainer):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::frameOwnerElement):
* Source/WebCore/page/SpatialNavigation.h:
(WebCore::FocusCandidate::inScrollableContainer const):
(WebCore::FocusCandidate::isFrameOwnerElement const): Deleted.

Canonical link: <a href="https://commits.webkit.org/273901@main">https://commits.webkit.org/273901@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/771e7be6649b3ffc945febad8c43f29e2773bf2d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37101 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16006 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39398 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39678 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33135 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38394 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18567 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13144 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31642 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37664 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13498 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32682 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11771 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11772 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/33192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40934 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33592 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37673 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12085 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9881 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35814 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13727 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8386 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12458 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/12991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/dom/Node.cpp:
```diff
@@ -2079,8 +2079,8 @@ static void showSubTreeAcrossFrame(const Node* node, const Node* markedNode, con
-        if (node->isFrameOwnerElement())
-            showSubTreeAcrossFrame(static_cast<const HTMLFrameOwnerElement*>(node)->contentDocument(), markedNode, indent + "\t");
+        if (auto* frameOwner = dynamicDowncast<HTMLFrameOwnerElement>(node))
+            showSubTreeAcrossFrame(frameOwner->contentDocument(), markedNode, indent + "\t");
```
I could see that we might want to use protectedContentDocument() here. If #23506 looks good and can land, I can rebase this on top of that and make that change.

cdumez replied:
Yes, let's leverage `protectedContentDocument()` here.

annevk commented on Source/WebCore/page/FocusController.cpp:
```diff
@@ -1113,13 +1113,13 @@ void FocusController::findFocusCandidateInContainer(Node& container, const Layou
 
     unsigned candidateCount = 0;
-    for (; element; element = (element->isFrameOwnerElement() || canScrollInDirection(element, direction))
+    for (; element; element = (is<HTMLFrameOwnerElement>(element) || canScrollInDirection(element, direction))
```
It reports a style error for this line, but that is not code I'm technically changing. Happy to take suggestions though.

cdumez replied:
Should be `is<HTMLFrameOwnerElement>(*element)`. You added an unnecessary null check.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -432,9 +432,8 @@ void InspectorDOMAgent::unbind(Node& node)
-        const HTMLFrameOwnerElement* frameOwner = static_cast<const HTMLFrameOwnerElement*>(&node);
-        if (Document* contentDocument = frameOwner->contentDocument())
+    if (auto* frameOwner = dynamicDowncast<HTMLFrameOwnerElement>(node)) {
+        if (auto* contentDocument = frameOwner->contentDocument())
```
RefPtr?

cdumez commented on Source/WebCore/page/FocusController.cpp:
```diff
@@ -1113,13 +1113,13 @@ void FocusController::findFocusCandidateInContainer(Node& container, const Layou
             continue;
 
-        if (!element->isKeyboardFocusable(event) && !element->isFrameOwnerElement() && !canScrollInDirection(element, direction))
+        if (!element->isKeyboardFocusable(event) && !is<HTMLFrameOwnerElement>(element) && !canScrollInDirection(element, direction))
```
`is<HTMLFrameOwnerElement>(*element)`

cdumez reviewed (approved):
LGTM with nits.

annevk committed:
Minor HTMLFrameOwnerElement cleanup
https://bugs.webkit.org/show_bug.cgi?id=268453

Reviewed by Chris Dumez.

This makes the following changes:

- Move to dynamicDowncast<HTMLFrameOwnerElement> where applicable.
- Move to is<HTMLFrameOwnerElement> where applicable.
- Rename isFrameOwnerElement() to isHTMLFrameOwnerElement() for
  clarity.

* Source/WebCore/dom/Node.cpp:
(WebCore::showSubTreeAcrossFrame):
* Source/WebCore/dom/Node.h:
(WebCore::Node::isHTMLFrameOwnerElement const):
(WebCore::Node::isFrameOwnerElement const): Deleted.
* Source/WebCore/html/HTMLFrameOwnerElement.h:
(isType):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::unbind):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::findFocusCandidateInContainer):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::frameOwnerElement):
* Source/WebCore/page/SpatialNavigation.h:
(WebCore::FocusCandidate::inScrollableContainer const):
(WebCore::FocusCandidate::isFrameOwnerElement const): Deleted.

Canonical link: https://commits.webkit.org/273901@main
